futures-util/src/io/buf_reader.rs

```diff
@@ -73,6 +74,43 @@ impl<R: AsyncRead> BufReader<R> {
     }
 }
 
+impl<R: AsyncRead + AsyncSeek> BufReader<R> {
+    /// Seeks relative to the current position. If the new position lies within the buffer,
+    /// the buffer will not be flushed, allowing for more efficient seeks.
+    /// This method does not return the location of the underlying reader, so the caller
+    /// must track this information themselves if it is required.
+    pub fn seek_relative(&mut self, offset: i64) -> SeeKRelative<'_, R>
```

Comment:
hmm, should this be `self: Pin<&mut Self>`?

Reply:
Oh, I cannot call seek_relative after creating BufReader if `R: !Unpin` and seek_relative needs `R: Unpin`.